### PR TITLE
[Bug Fix] Fix language model warmup logic error

### PR DESCRIPTION
**Pull request by classicsong**
The intend of freeze_lm_encoder_epochs is to freeze language model for several epochs, but the implementation freeze the model for several iterations.

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: python/graphstorm/config/argument.py [hunk 1]
```diff
@@ -187,6 +189,22 @@ def override_arguments(self, cmd_args):
                 setattr(self, f"_{arg_key}", arg_val)
                 print(f"Overriding Argument: {arg_key}")
 
+    def handle_argument_conflicts(self):
+        """Check and resolve argument conflicts
+        """
+        # 1. language model conflicts
+        if self.node_lm_configs is not None:
+            # gradient checkpoint does not work with freeze_lm_encoder_epochs
+            # When freeze_lm_encoder_epochs is set, turn off gradient checkpoint
```
Comment (zheng-da) [inline python/graphstorm/config/argument.py:198]: if gradient checkpoint is enabled, we cannot freeze the model training of the BERT models?

File: python/graphstorm/config/argument.py [hunk 2]
```diff
@@ -305,6 +323,12 @@ def freeze_lm_encoder_epochs(self):
         if hasattr(self, "_freeze_lm_encoder_epochs"):
             assert self._freeze_lm_encoder_epochs >= 0, \
                 "Number of warmup epochs must be larger than or equal to 0"
+
+            assert self._freeze_lm_encoder_epochs == 0 or \
+                self.model_encoder_type not in ["lm", "mlp"], \
```
Comment (zheng-da) [inline python/graphstorm/config/argument.py:328]: even MLP doesn't work?

File: python/graphstorm/trainer/ep_trainer.py [hunk 3]
```diff
@@ -112,6 +112,9 @@ def fit(self, train_loader, n_epochs,
         for epoch in range(n_epochs):
             model.train()
             t0 = time.time()
+            if freeze_input_layer_epochs <= epoch:
+                self._model.unfreeze_input_encoder()
+            # TODO(xiangsx) Support unfreezing gnn encoder and decoder
```
Comment (zheng-da) [inline python/graphstorm/trainer/ep_trainer.py:117]: we need to have a regression test to verify the BERT+GNN model training.

Comment (classicsong) [reply python/graphstorm/trainer/ep_trainer.py:117]: We do.

**Review by zheng-da — APPROVED** [review]
